components/authz-service/server/v2/policy_refresher.go

```diff
@@ -162,10 +153,16 @@ func (refresher *policyRefresher) RefreshAsync() error {
 }
 
 // updates OPA engine store with policy
-func (refresher *policyRefresher) updateEngineStore(ctx context.Context, vsn api.Version) error {
+func (refresher *policyRefresher) updateEngineStore(ctx context.Context) error {
 	// Engine updates need unfiltered access to all data.
 	ctx = auth_context.ContextWithoutProjects(ctx)
 
+	vsn, err := refresher.getIAMVersion(ctx)
```

Comment:
Not following all the intricacies of this PR, but I think this line is the crux: it retrieves the version from the DB instead of the local server state so that in a multi-node situation it will be updating the correct store. Is that right? If so, please add an in-code comment to that effect; thx!